### PR TITLE
Adding github workflows

### PR DESCRIPTION
**Pull request by TrangPham**
Tested on my fork: https://github.com/TrangPham/jsonforms-antd-renderers/actions/runs/8089417491/job/22105275009

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
```
Comment (DrewHoo) [inline .github/workflows/tests.yaml:22]: There's a way to reference the version specified in the .nvmrc--see mercury-web's deploy or validate workflows for an example of the exact syntax

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        run: |
+          echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV
+
+      - uses: actions/cache@v3
+        name: Setup pnpm cache
+        with:
+          path: ${{ env.STORE_PATH }}
+          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-pnpm-store-
```
Comment (DrewHoo) [inline .github/workflows/tests.yaml:41]: Trailing dash here seems odd

File: .github/workflows/tests.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+      - name: Install dependencies
+        run: pnpm install
```
Comment (DrewHoo) [inline .github/workflows/tests.yaml:44]: Does this use the lockfile by default or do you need to specify frozen-lockfile?

**Review by DrewHoo — CHANGES_REQUESTED** [review]

**Review by NathanFarmer — COMMENTED** [review]
Are we planning to use the pre-commit stage? Was that added to our entire great-expectations organization or did we add that?

**Comment by Kilo59** [comment]
Just opened a quick PR with some minimal pre-commit config settings
https://github.com/great-expectations/jsonforms-antd-renderers/pull/15

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (DrewHoo) [reply .github/workflows/tests.yaml:22]: https://github.com/greatexpectationslabs/mercury-web/blob/main/.github/actions/setup-node/action.yml

File: .github/workflows/tests.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
+name: tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```
Comment (DrewHoo) [inline .github/workflows/tests.yaml:9]: I don't think this is an issue but want to double check: if joe schmoe opens a PR against this repo, can they do stuff like make changes to this workflow that logs api keys or anything like that? I'm pretty sure the answer is no

**Review by DrewHoo — APPROVED** [review]

**Pull request merged**